sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatch.cs

```diff
@@ -0,0 +1,294 @@
+        public virtual Response DeleteBlob(
+            Uri blobUri,
+            DeleteSnapshotsOption? deleteOptions = default,
+            BlobAccessConditions? accessConditions = default)
+        {
+            EnsureAllowed(BlobBatchOperationType.Delete);
+            HttpPipelineMessage message = BlobRestClient.Blob.DeleteAsync_CreateMessage(
+                _client.BatchOperationPipeline,
+                blobUri,
+                deleteSnapshots: deleteOptions,
+                leaseId: accessConditions?.LeaseAccessConditions?.LeaseId,
+                ifModifiedSince: accessConditions?.HttpAccessConditions?.IfModifiedSince,
+                ifUnmodifiedSince: accessConditions?.HttpAccessConditions?.IfUnmodifiedSince,
+                ifMatch: accessConditions?.HttpAccessConditions?.IfMatch,
+                ifNoneMatch: accessConditions?.HttpAccessConditions?.IfNoneMatch);
+            _messages.Add(message);
+            return new DelayedResponse(message, BlobRestClient.Blob.DeleteAsync_CreateResponse);
+        }
+        #endregion DeleteBlob
+
+        #region SetBlobAccessTier
+        /// <summary>
+        /// The <see cref="SetBlobAccessTier(string, string, AccessTier, RehydratePriority?, LeaseAccessConditions?)"/>
+        /// operation sets the tier on a blob.  The operation is allowed on
+        /// block blobs in a blob storage or general purpose v2 account.
+        ///
+        /// A block blob's tier determines Hot/Cool/Archive storage type.  This
+        /// operation does not update the blob's ETag.  For detailed
+        /// information about block blob level tiering
+        /// <see href="https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers" />.
+        /// </summary>
+        /// <param name="containerName">
+        /// The name of the container containing the blob to set the tier of.
+        /// </param>
+        /// <param name="blobName">
+        /// The name of the blob to set the tier of.
+        /// </param>
+        /// <param name="accessTier">
+        /// Indicates the tier to be set on the blob.
+        /// </param>
+        /// <param name="leaseAccessConditions">
+        /// Optional <see cref="LeaseAccessConditions"/> to add conditions on
+        /// setting the access tier.
+        /// </param>
+        /// <param name="rehydratePriority">
+        /// Optional <see cref="RehydratePriority"/>
+        /// Indicates the priority with which to rehydrate an archived blob.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.  The response
+        /// cannot be used until the batch has been submitted with
+        /// <see cref="BlobBatchClient.SubmitBatchAsync"/>.
+        /// </returns>
+        public virtual Response SetBlobAccessTier(
+            string containerName,
+            string blobName,
+            AccessTier accessTier,
+            RehydratePriority? rehydratePriority = default,
```

Comment:
Why nullable (both this and leaseAccessConditions)?

Reply:
Ditto on the `BlobBaseClient.SetAccessTier` having these as nullable.

Reply:
Created #7974 to track the issue.